tests/testversion.py

```diff
@@ -160,6 +165,32 @@ def test_git_describe_4(self):
     def test_git_describe_5(self):
         self.git_describe_check(describe_tests, 5)
 
+    #=======================================#
+    #  Compatibility with ASGI applications #
+    #=======================================#
+
+    def test_run_asgi_server(self):
+        """test_run_asgi_server (An ASGI app that uses autover and is launched with Gunicorn should not crash)"""
+        if sys.version[0] == "3":
+            serverfile_dir = os.path.abspath(os.path.join(__file__, os.pardir))
+            logfile_path = os.path.abspath(os.path.join(__file__, os.pardir, "stderr_gunicorn.log"))
+            subprocess.Popen(["gunicorn", "-D", "--log-level", "ERROR", 
+                              "--chdir", serverfile_dir,
+                              "-k", "uvicorn.workers.UvicornWorker", 
+                              "--error-logfile", logfile_path,
+                              "--bind", "unix:/tmp/gunicorn.sock", 
+                              "test_asgi_server:app"])
+            time.sleep(5)
+            with open(logfile_path, "r") as gunicorn_log_file:
+                stderr_log = gunicorn_log_file.read()
+
+            subprocess.Popen(["pkill", "-f", "gunicorn"])
```

Comment:
I guess it would be better to kill exactly the pid that was created, in case there are other things running. But we might be able to create an entirely simpler test (we'll see once I can reproduce), so let's wait and see.

Reply:
Yeah, I guess you are right. Fortunately, Gunicorn can output the PID that launches, so can be retreived by Python and be killed.